torch/csrc/autograd/profiler.cpp

```diff
@@ -418,9 +485,62 @@ void Event::record(bool record_cuda) {
   cpu_ns_ = getTime();
 }
 
-double Event::cuda_elapsed_us(const Event & e) {
+/* static */ Event Event::fromIValue(const at::IValue& eventIValue) {
+  TORCH_INTERNAL_ASSERT(
+      eventIValue.isList(),
+      "Expected IValue to contain type c10::impl::GenericList");
+  auto ivalues = eventIValue.toList();
+  TORCH_INTERNAL_ASSERT(
+      ivalues.size() >= kEventIValuesSize,
+      "Expected at least ",
+      kEventIValuesSize,
+      " elements to reconstruct Event.");
+
+  Event evt(
+    static_cast<EventKind>(ivalues.get(0).toInt()), // EventKind
+    at::StringView(ivalues.get(1).toStringRef()), // name
+    ivalues.get(2).toInt(), // thread
+    static_cast<at::RecordFunctionHandle>(ivalues.get(3).toDouble()), // handle
+    {}, // TODO: record shapes
+    ivalues.get(4).toInt(), // node id
+    true, // is remote
+    ivalues.get(5).toInt(), // cpu_mem_usage
```

Comment:
nit: linter actually makes sense - maybe use enum to name position constants?

Reply:
(up to you)

Reply:
Sounds good, I'll update to use an enum